index/repo.go

```diff
@@ -0,0 +1,50 @@
+package index
+
+import (
+	"github.com/filecoin-project/dagstore/shard"
+)
+
+// Repo is the central index repository object that manages full indices and
+// manifests.
+type Repo struct {
+	FullIndexRepo
+	ManifestRepo
+}
+
+type Stat struct {
+	Exists bool
+	Size   uint64
+}
+
+type FullIndexRepo interface {
```

Comment:
I suggest we include methods to
- get the count of indexes in the repo
- iterate over index keys
- get the size of the repo

Reply:
Agree, wanna push a change here?

Reply:
These changes are pushed as part of #16.